### PR TITLE
Updated account settings for push notifications

### PR DESCRIPTION
**Pull request by yangchen1**
Doc for https://mattermost.atlassian.net/browse/PLT-3399, https://mattermost.atlassian.net/browse/PLT-3462, and https://mattermost.atlassian.net/browse/PLT-3461.

### CODE REVIEW

File: source/help/settings/account-settings.md [hunk 1]
```diff
@@ -62,7 +62,9 @@ A notification sound plays for all Mattermost posts that would fire a desktop no
 Email notifications are sent for mentions and direct messages after you’ve been offline for more than 60 seconds or away from Mattermost for more than 5 minutes. Change the email where notifications are sent from **Account Settings > General > Email**.
 
 #### Mobile Push Notifications
-Push notifications are sent to your mobile device if you have the Android or iOS app installed. They are sent even if you are online, similar to desktop notifications. By default, push notifications are sent "For mentions and direct messages". If your System Administrator has not set up push notifications, this setting will be disabled.
+Push notifications can be sent to your mobile device if you have the Android or iOS app installed. When they are sent can be adjusted depending on the type of notification and your status. By default, push notifications are sent "For mentions and direct messages", and if your status is "Online, away or offline". If push notifications are sent "Never", the setting to trigger push notifications depending on your status is hidden. If your System Administrator has not set up push notifications, both settings will be disabled.
```
Comment (lfbrock) [inline source/help/settings/account-settings.md]: Can you clarify this sentence? "When they are sent can be adjusted depending on the type of notification and your status."

Or maybe we should break it into two sections to describe each setting.

File: source/help/settings/account-settings.md [hunk 2]
```diff
@@ -62,7 +62,9 @@ A notification sound plays for all Mattermost posts that would fire a desktop no
...
+
+Also, if you are actively viewing a channel (public, private, or direct message) on desktop or web, no push notifications will be sent for that channel. If a notification is viewed on desktop or web, any push notifications and app icon badges already sent will be cleared.
```
Comment (lfbrock) [inline source/help/settings/account-settings.md]: I don't think push notifications are cleared for iOS yet (badge icon should be cleared though I think, you can confirm with Eric)

File: source/help/settings/account-settings.md (hunk 1 above)
Comment (esethna) [inline source/help/settings/account-settings.md]: Agree, I think we should have a separate section for each. ie see my [PR](https://github.com/mattermost/docs/pull/477/files) for desktop settings.

image

File: source/help/settings/account-settings.md (hunk 2 above)
Comment (esethna) [inline source/help/settings/account-settings.md]: Ya agree, please separate this describing what happens on iOS vs Android. iOS only clears the red badge on the Mattermost icon, not the lockscreen. Maybe something like:

`If a notification is viewed on desktop or web, the lockscreen notification will clear on Android, and on iOS the badge on the Mattermost icon will decrement accordingly.`

**Comment by esethna** [comment]
Also please add a section for push notifications int he configuring-notifications.md document

File: source/help/getting-started/configuring-notifications.md [hunk 3]
```diff
@@ -20,11 +20,17 @@ Emails are sent for any mentions you receive if you have mattermost closed or ha
...
+These are browser notifications that appear in the corner of your main monitor. By default, these notifications are sent for all activity while Mattermost is open. Desktop notifications are available on Firefox, Safari, and Chrome.
 
-- Configure when desktop notifications are sent from **Account Settings** > **Notifications** > **Send Desktop Notifications**.
+- Configure when desktop notifications are sent from **Account Settings** > **Notifications** > **Desktop Notifications** > **Send Desktop Notifications**.
 - Configure channel specific desktop notifications from **Channel Menu** > **Notification Preferences** > **Send Desktop Notifications**. By default, all channels use the global setting configured in *Account Settings*.
 
+## Mobile Push Notifications
+If the Mattermost Android or iOS app is installed, push notifications can be sent to your mobile device. By default, these notifications are sent for all activity for any channel that is not being viewed on desktop.
```
Comment (jasonblais) [inline source/help/getting-started/configuring-notifications.md]: I think we send push notifications for mentions and DMs by default

File: source/help/settings/account-settings.md [hunk 4]
```diff
@@ -61,8 +61,13 @@ A notification sound plays for all Mattermost posts that would fire a desktop no
 #### Email Notifications
 Email notifications are sent for mentions and direct messages after you’ve been offline for more than 60 seconds or away from Mattermost for more than 5 minutes. Change the email where notifications are sent from **Account Settings > General > Email**.
 
-#### Mobile Push Notifications
-Push notifications are sent to your mobile device if you have the Android or iOS app installed. They are sent even if you are online, similar to desktop notifications. By default, push notifications are sent "For mentions and direct messages". If your System Administrator has not set up push notifications, this setting will be disabled.
+#### Send Mobile Push Notifications
+Push notifications can be sent to your mobile device if you have the Android or iOS app installed. You can choose the type of activity that will send a notification. By default, push notifications are sent "For mentions and direct messages". If push notifications are sent "Never", the setting to trigger push notifications depending on your status is hidden. If your System Administrator has not set up push notifications, this setting will be disabled.
```
Comment (jasonblais) [inline source/help/settings/account-settings.md]: "the setting to trigger push notifications depending on your status is hidden"

in Mattermost settings or device settings?

File: source/help/getting-started/configuring-notifications.md (hunk 3 above)
Comment (yangchen1) [inline source/help/getting-started/configuring-notifications.md]: Agreed! Do we need to file a separate JIRA ticket for that on platform?

Comment (jasonblais) [inline source/help/getting-started/configuring-notifications.md]: I think it's by design to send push notifications for mentions only (by default), so we just need to update it in the doc. 

Maybe confirm with @esethna as he owns notifications?

Comment (yangchen1) [inline source/help/getting-started/configuring-notifications.md]: updated!

File: source/help/settings/account-settings.md (hunk 4 above)
Comment (yangchen1) [inline source/help/settings/account-settings.md]: updated!

**Comment by esethna** [comment]
+1, @yangchen1 please resolve the conflicts then we can merge?

**Pull request merged**